polkadot/node/network/statement-distribution/src/v2/mod.rs

```diff
@@ -2138,28 +2137,34 @@ async fn provide_candidate_to_grid<Context>(
 }
 
 // Utility function to populate per relay parent `ParaId` to `GroupIndex` mappings.
-fn determine_groups_per_para(
+async fn determine_groups_per_para(
+	sender: &mut impl overseer::StatementDistributionSenderTrait,
+	relay_parent: Hash,
 	availability_cores: Vec<CoreState>,
 	group_rotation_info: GroupRotationInfo,
-	maybe_claim_queue: Option<ClaimQueueSnapshot>,
 	max_candidate_depth: usize,
 ) -> HashMap<ParaId, Vec<GroupIndex>> {
+	let maybe_claim_queue = fetch_claim_queue(sender, relay_parent)
```

Comment:
Ok, my thinking was to move all fetches here, including `availability_cores` for consistency. It's weird if we pass in one thing, but fetch the other our selves. I know this is used in one other place as well (wrongly), but given that we pass per value already, there is not much gain in avoiding the double fetch.

Reply:
LocalValidatorState is built outside this fn and requires `availability_cores`